temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java

```diff
@@ -41,6 +53,34 @@
       doNotRetry = {"java.lang.NullPointerException", "java.lang.UnsupportedOperationException"})
   public void activityAndRetryOptions() {}
 
+  @Test
+  public void testActivityOptions() {
+    testEnv.registerActivitiesImplementations(new TestActivityImpl());
+    try {
+      testEnv.newActivityStub(TestActivity.class);
+    } catch (IllegalArgumentException e) {
+      Assert.assertTrue(e instanceof IllegalArgumentException);
+      Assert.assertEquals(
+          e.getMessage(),
+          "Both StartToCloseTimeout and ScheduleToCloseTimeout aren't specified for Activity1 activity. Please set at least one of the above through the ActivityStub or WorkflowImplementationOptions.");
+    }
+  }
+
+  @Test
+  public void testActivityOptionsDefaultInstance() {
+    testEnv.registerActivitiesImplementations(new TestActivityImpl());
+    TestActivity activity =
+        testEnv.newActivityStub(
+            TestActivity.class,
+            ActivityOptions.newBuilder().setScheduleToCloseTimeout(Duration.ofDays(1)).build(),
+            null);
+
+    // Check that options were set correctly
+    Map<String, Duration> optionsValues = activity.activity1();
+    Assert.assertEquals(Duration.ofDays(1), optionsValues.get("ScheduleToCloseTimeout"));
+    Assert.assertEquals(Duration.ofDays(1), optionsValues.get("StartToCloseTimeout"));
```

Comment:
I"m confused. Where does this timeout come from if it is not set?

Reply:
I wasn't sure about the reasoning behind this logic, but this is where it comes from:
https://github.com/temporalio/sdk-java/blob/4c9bdd9c445360c0b3cffb080fb8e28527910e70/temporal-sdk/src/main/java/io/temporal/activity/ActivityOptions.java#L256

Reply:
That logic seems outdated. We didn't have sane defaults on the service some time ago, so it was introduced. Let's remove it from the ActivityOptions completely.